examples/appSpec/src/main.rs

```diff
@@ -0,0 +1,84 @@
+#[macro_use]
+extern crate serde_derive;
+
+extern crate holochain_core;
+extern crate holochain_core_api;
+extern crate holochain_dna;
+extern crate test_utils;
+
+#[cfg(test)]
+mod tests {
+    extern crate serde;
+    extern crate serde_json;
+
+    use std::fs::File;
+    use std::collections::HashMap;
+    use std::io::prelude::*;
+    use holochain_dna::*;
+    use holochain_dna::zome::Zome;
+    use holochain_core_api::*;
+    use test_utils::*;
+
+    #[derive(Deserialize, Debug)]
+    struct CommitResult {
+        hash: String
+    }
+
+    #[test]
+    fn testing() {
+        // Setup the holochain instance
+        let mut dna_file = File::open("app.hcpkg").expect("file not found");
+        let mut dna_string = String::new();
+        dna_file.read_to_string(&mut dna_string)
+            .expect("something went wrong reading the file");
+
+        let mut dna = Dna::from_json_str(&dna_string).unwrap();
+
+        // We need to inject a capability with empty string as name because the validation callback
+        // has set its capability to nothing and the callback mechanism is using that as a string
+        // and tries to call the callback there.
+        // TODO:
+        // That has to be changed. Validation callbacks should be found in the WASM of the entry type
+        // instead.
+        // Or we go all the way and change the spec to have only one WASM module per zome..
+        // See: https://github.com/holochain/holochain-rust/issues/342
+        dna.zomes = dna
```

Comment:
later on we should work on getting valid asm based `validate_commit` callback working, but for now this works